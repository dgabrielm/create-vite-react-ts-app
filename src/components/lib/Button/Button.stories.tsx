import { Meta, StoryObj } from '@storybook/react'
import Button from 'components/lib/Button'
import 'scss/globals.scss'

const meta: Meta<typeof Button> = {
  title: 'Components/Button',
  component: Button,
  argTypes: { handleClick: { action: 'handleClick' } },
  parameters: {
    layout: 'centered'
  }
}

export default meta

type Story = StoryObj<typeof Button>

export const Demo: Story = {
  args: {
    label: 'Press me'
  }
}
