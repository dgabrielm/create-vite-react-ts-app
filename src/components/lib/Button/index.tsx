import React, { MouseEventHandler } from 'react'
import styles from './styles.module.scss'

interface ButtonProps {
  label: string
  handleClick: MouseEventHandler
}

function Button({ label, handleClick }: ButtonProps) {
  return (
    <button
      className={styles.button}
      onClick={handleClick}
    >
      {label}
    </button>
  )
}

export default Button
export type { ButtonProps }
