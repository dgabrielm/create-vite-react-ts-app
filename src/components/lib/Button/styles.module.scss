@use 'scss/abstracts/_theme.module.scss' as theme;

.button {
  font-weight: 600;
  height: 40px;
  font-size: 15px;
  line-height: 22px;
  text-align: center;
  padding: 7px 14px;
  cursor: pointer;
  border-radius: 8px;
  color: theme.$primary_1;
  min-width: max-content;
  position: relative;

  & * {
    color: theme.$primary_1;
  }
}
