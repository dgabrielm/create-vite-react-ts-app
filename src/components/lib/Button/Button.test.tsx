import Button from '.'
import { describe, afterEach, expect, it } from 'vitest'
import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

describe('Button.tsx', () => {
  afterEach(() => {
    cleanup()
  })

  it('should render the page title', () => {
    render(<Button label="Press Me" handleClick={() => {}} />)
    const headingElement = screen.getByRole('button')
    expect(headingElement).toHaveTextContent('Press Me')
  })

  it('should match snapshot', () => {
    const tree = renderer.create(<Button label="Press Me" handleClick={() => {}} />)
    expect(tree).toMatchSnapshot()
  })
})
