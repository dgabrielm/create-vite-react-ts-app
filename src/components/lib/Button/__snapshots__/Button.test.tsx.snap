// Vitest Snapshot v1

exports[`Button.tsx > should match snapshot 1`] = `
<button
  className="_button_6dba28"
  onClick={[Function]}
>
  Press Me
</button>
`;
